### PR TITLE
enhance downloading APIs

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

- enhance the download APIs with better default path values
- remove duplicated check_hash in `download_and_extract`

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/utils.py [hunk 1]
```diff
@@ -37,6 +39,12 @@
 ]
 
 
+def _basename(p):
+    """get the last part of the path (removing the trailing slash if it exists)"""
+    sep = os.path.sep + (os.path.altsep or "") + "/ "
```
Comment (Nic-Ma) [inline monai/apps/utils.py:44]: Please double confirm whether it works fine for windows?
Thanks.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me except for minor comments.

Thanks.

File: monai/apps/utils.py (hunk 1 above)
Comment (wyli) [reply monai/apps/utils.py:44]: sure, and the '/' is mainly for getting filename from the URLs

**Pull request merged**